src/store/sagas/dataset-freestyle-query-saga.js

```diff
@@ -3,6 +3,9 @@ import { EXECUTE_FREESTYLE_QUERY, CANCEL_FREESTYLE_QUERY, setQuerySuccess, setQu
 import { getDatasetQueryCancelToken } from '../selectors'
 import axios from 'axios'
 
+const cancelMessage = 'Your query has been stopped'
```

Comment:
The changes in this file are probably the most noteworthy.

The failure message was only being used as a truthy boolean to trigger the display of constant error messages if the failure message in Redux was not set to empty string.

To avoid the indirection, I pulled the constant error messages and cancelled messages into the saga.

Reply:
I like this. We generally did this with the visualization stuff.